src/ts/background/services/keyring-vault.ts

```diff
@@ -23,6 +27,47 @@ class KeyringVault {
     throw new Error(t('keyringNotInit'))
   }
 
+  getTempPassword (): string {
+    return this._tempPassword || ''
+  }
+
+  setTempPassword (tempPassword: string): void {
+    this._tempPassword = tempPassword
+
+    // Trigger clear interval for tempPassword
+    this.clearTempPassword()
+  }
+
+  clearTempPassword (): void {
+    this.startExpiryTimer()
```

Comment:
why do we need to start expiry timer inside clear temp password func?

Reply:
Ahh. We don't. I can move it up to `setTempPassword()` real quick. Thanks for pointing it out.